### PR TITLE
Added validation for the new clusters field.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
1. Added validation for the clusters field. 
2. Refactored ClusterMetricsInput validiation to throw 4xx-level CommonUtilsExceptions instead of 5xx-level IllegalArgumentException.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by engechas — DISMISSED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt [hunk 1]
```diff
@@ -15,6 +16,14 @@ import java.io.IOException
...
 import java.net.URISyntaxException
 
+/**
+ * This regex asserts that the string:
+ *  Starts with a lowercase letter, or digit
+ *  Contains a sequence of characters followed by an optional colon and another sequence of characters
+ *  The sequences of characters can include lowercase letters, uppercase letters, digits, underscores, or hyphens
+ *  The total length of the string can range from 1 to 255 characters
+ */
+val CLUSTER_NAME_REGEX = Regex("^(?=.{1,255}$)[a-z0-9]([a-zA-Z0-9_-]*:?[a-zA-Z0-9_-]*)$")
```
[the remark this answers is not in the document]
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt:26]: As discussed offline, I haven't come across documentation which details cluster name restrictions; I've reached out to the core team to get more insight. I did come across this line of validation code.
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/ClusterName.java#L59

We can loosen up these restrictions as needed in the future.

Comment (engechas) [inline src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt:26]: Let's remove this and reference the one in IndexUtils

File: src/main/kotlin/org/opensearch/commons/alerting/util/IndexUtils.kt [hunk 2]
```diff
@@ -11,6 +11,33 @@ import java.time.Instant
...
+        val INDEX_PATTERN_REGEX = Regex("""^(?=.{1,255}$)\.?[a-z0-9_\-\*\.]+$""")
+
+        /**
+         * This regex asserts that the string:
+         *  Starts with a lowercase letter, digit, or asterisk
+         *  Contains a sequence of characters followed by an optional colon and another sequence of characters
+         *  The sequences of characters can include lowercase letters, uppercase letters, digits, underscores, asterisks, or hyphens
+         *  The total length of the string can range from 1 to 255 characters
+         */
+        val CLUSTER_PATTERN_REGEX = Regex("^(?=.{1,255}$)[a-z0-9*]([a-zA-Z0-9_*-]*:?[a-zA-Z0-9_*-]*)$")
```
Comment (engechas) [inline src/main/kotlin/org/opensearch/commons/alerting/util/IndexUtils.kt:39]: Non-blocking - Really minor: this isn't really related to indices so IndexUtils is probably not the right home. `ValidationHelpers` would be better

File: src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt (hunk 1 above)
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt:26]: Oh good catch. Will do.

File: src/main/kotlin/org/opensearch/commons/alerting/util/IndexUtils.kt (hunk 2 above)
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/util/IndexUtils.kt:39]: That makes sense. I'm revising this PR anyway, so I'll add it to a `ValidationHelpers` file.

**Review by engechas — APPROVED** [review]

**Review by riysaxen-amzn — APPROVED** [review]

**Pull request merged**